src/libs/workspace-utils.ts

```diff
@@ -159,10 +159,26 @@ export const isValidWsExportTarget = safeCurry((sourceWs: WorkspaceWrapper, dest
   );
 });
 
-// Returns a message explaining why the user can't edit the workspace, or undefined if they can
-export const editWorkspaceError = ({ accessLevel, workspace: { isLocked } }) => {
-  return cond(
-    [!canWrite(accessLevel), () => 'You do not have permission to modify this workspace.'],
-    [isLocked, () => 'This workspace is locked.']
+export const canEditWorkspace = ({
+  accessLevel,
+  workspace: { isLocked },
+}: {
```

Comment:
Seems this type would be good to export above the multiple uses here, as something like
```
export interface WorkspaceAccessInfo {
  accessLevel: WorkspaceAccessLevel;
  workspace: { isLocked: boolean }
}

```
Then just reference the type in the function signatures here.

Reply:
https://github.com/DataBiosphere/terra-ui/pull/4268/commits/0cd34e362bca1a253c51b4b27701a724f67c9605